core/tests/lightclient.rs

```diff
@@ -60,7 +64,82 @@ mod test {
         assert_eq!(bytes_result_value, result_value);
     }
 
-    fn config() -> (MockStarknetClient, StarknetClientImpl) {
+    #[tokio::test]
+    async fn when_get_block_with_tx_hashes_then_should_return_bytes() {
+        // Given
+        let mut starknet_lightclient_mock = config();
+
+        let hash = H256::default();
+        let parent_hash = H256::default();
+        let sequencer = H160::default();
+        let state_root = H256::default();
+        let number = U256::from(100);
+        let timestamp = U256::from(100);
+        let transactions = BlockTransactions::Hashes(vec![]);
+        let gas_limit = U256::ZERO;
+        let gas_used = U256::ZERO;
+        let difficulty = U256::ZERO;
+        let nonce: Option<H64> = None;
+        let size: Option<U256> = None;
+        // Bloom is a byte array of length 256
+        let logs_bloom = Bloom::default();
+        let extra_data = Bytes::from(b"0x00");
+        let total_difficulty: U256 = U256::ZERO;
+        let header = Header {
+            hash: Some(hash),
+            parent_hash,
+            uncles_hash: parent_hash,
+            author: sequencer,
+            miner: sequencer,
+            state_root,
+            // BlockWithTxHashes doesn't have a transactions root
+            transactions_root: H256::zero(),
+            // BlockWithTxHashes doesn't have a receipts root
+            receipts_root: H256::zero(),
+            number: Some(number),
+            gas_used,
+            gas_limit,
+            extra_data,
+            logs_bloom,
+            timestamp,
+            difficulty,
+            nonce,
+            size,
+        };
+        let block = Block {
+            header,
+            total_difficulty,
+            uncles: vec![],
+            transactions,
+            base_fee_per_gas: None,
+            size,
+        };
+        let expected_result = Rich::<Block> {
+            inner: block,
+            extra_info: BTreeMap::default(),
+        };
+
+        let expected_result_value = expected_result.clone();
+
+        starknet_lightclient_mock
```

Comment:
isn't this test testing that a clone of `expected_result` is equal to another clone of `expected_result`?

Reply:
I think what this test might need to perform is:
Construct a "dummy" StarknetBlock with txHashes and with tx
then pass them inside `from_starknet_transaction_ethers_transaction`

and assert that the returned transaction is an ethereum tx

Reply:
Oh no, I am just using expect_result_value, because expected_result ownership moves when we do the mocking, so its just a clone to compare the end result with "expected_result" 

I believe for the second point would be to test "from_starknet_transaction_ethers_transaction", but I think I this should be another test, so I will add one there.

Reply:
Yes makes sense, I'm just saying `.expect_get_eth_block_from_starknet_block()
            .returning(move |_, _| Ok(expected_result.clone()));` makes it so that the `.get_eth_block_from_starknet_block` method does nothing except return expected_result.clone(), which means that we're not testing anything here;)

Reply:
Yes, this is how automock works. In case we wanted to test the real implementation, we would need an RPC Link and the tests would be a bit different

Reply:
My point is that this test is not testing anything, we can delete it and instead test helpers functions:)!

Then, we only have to make sure that the starknet_client `get_eth_block_from_starknet_block()` has no logic beyond assembling/rearranging our helpers functions

Reply:
Ok! May I implement this change on the next PR? I will also add the tests for the helper functions

Reply:
Awesome!